reactor-core/src/main/java/reactor/core/publisher/FluxTakeWhile.java

```
@@ -96,6 +100,9 @@ public void onNext(T t) {
 
 		@Override
 		public void onError(Throwable t) {
+			if (took) {
+				return;
```

same here, I don't think the throwable should be silently ignored